smp/enumeration_management.py

```diff
@@ -0,0 +1,123 @@
+"""The Simple Management Protocol (SMP) Enumeration Management group."""
+
+from __future__ import annotations
+
+from enum import IntEnum, unique
+from typing import Tuple
+
+from pydantic import BaseModel, ConfigDict
+
+import smp.error as smperr
+import smp.header as smphdr
+import smp.message as smpmsg
+
+
+class GroupCountRequest(smpmsg.ReadRequest):
+    """Read the number of SMP server groups."""
+
+    _GROUP_ID = smphdr.GroupId.ENUM_MANAGEMENT
+    _COMMAND_ID = smphdr.CommandId.EnumManagement.GROUP_COUNT
+
+
+class GroupCountResponse(smpmsg.ReadResponse):
+    """SMP group count response."""
+
+    _GROUP_ID = smphdr.GroupId.ENUM_MANAGEMENT
+    _COMMAND_ID = smphdr.CommandId.EnumManagement.GROUP_COUNT
+
+    count: int
+
+
+class ListOfGroupsRequest(smpmsg.ReadRequest):
+    """List the available SMP groups."""
+
+    _GROUP_ID = smphdr.GroupId.ENUM_MANAGEMENT
+    _COMMAND_ID = smphdr.CommandId.EnumManagement.LIST_OF_GROUPS
+
+
+class ListOfGroupsResponse(smpmsg.ReadResponse):
+    """SMP group list response."""
+
+    _GROUP_ID = smphdr.GroupId.ENUM_MANAGEMENT
+    _COMMAND_ID = smphdr.CommandId.EnumManagement.LIST_OF_GROUPS
+
+    groups: Tuple[int, ...]
```

Comment:
https://github.com/JPHutchins/smp/blob/43ee844ddaf5bcdb8dda2b81c52a95438833aa69/smp/message.py#L31C25-L31C89

Reply:
Yeah, that seems like it would be less confusing! Try to match IntEnum first as well in case the user added IntEnums.

```
GroupId | UserGroupId | IntEnum | int
```